ir/instr.cpp

```diff
@@ -3603,23 +3613,55 @@ void ShuffleVector::rauw(const Value &what, Value &with) {
 }
 
 void ShuffleVector::print(ostream &os) const {
-  os << getName() << " = shufflevector " << *v1 << ", " << *v2;
-  for (auto m : mask)
-    os << ", " << m;
+  const char *str = nullptr;
+  switch(op) {
+  case LLVMIR_ShufVec:   str = "shufflevector"; break;
+  case SSSE3_PShufB128:  str = "ssse3.pshuf.b.128"; break;
+  case AVX2_PShufB:      str = "avx2.pshuf.b"; break;
+  case AVX512_PShufB512: str = "avx512.pshuf.b.512"; break;
+  }
+  os << getName() << " = " << str << " " << *v1 << ", " << *v2;
+  if (op == LLVMIR_ShufVec)
+    for (auto m : mask)
+      os << ", " << m;
 }
 
 StateValue ShuffleVector::toSMT(State &s) const {
-  auto vty = v1->getType().getAsAggregateType();
+  auto vty = static_cast<const VectorType*>(v1->getType().getAsAggregateType());
```

Comment:
we can't do a static cast on types because of Symbolic Types. And I don't think it's needed (see below).

Reply:
InsertElement::toSMT has the static cast too. Should I change it too?

Reply:
yes, please!

Reply:
I take this back, sorry. Since that operand can only be a vector, there's no issue. The problem is when the operand may have more than one type (e.g., int or vector of ints).

Reply:
Oh, okay. No problem.